chain-impl-mockchain/src/leadership/genesis/mod.rs

```diff
@@ -136,6 +136,137 @@ impl GenesisLeaderSelection {
     }
 }
 
+#[cfg(test)]
+mod test {
+
+    use super::*;
+    use crate::ledger::test::create_initial_fake_ledger;
+    use crate::milli::Milli;
+    use crate::stake::PoolStakeDistribution;
+    use crate::stake::StakePoolInfo;
+    use chain_addr::Discrimination;
+    use std::collections::HashMap;
+
+    fn make_pool(ledger: &mut Ledger) -> (StakePoolId, SecretKey<Curve25519_2HashDH>) {
+        let mut rng = rand::thread_rng();
+
+        let pool_vrf_private_key = SecretKey::generate(&mut rng);
+
+        let pool_info = StakePoolInfo {
+            serial: 1234,
+            owners: vec![],
+            initial_key: GenesisPraosLeader {
+                vrf_public_key: pool_vrf_private_key.to_public(),
+                kes_public_key: SecretKey::generate(&mut rng).to_public(),
+            },
+        };
+
+        let pool_id = pool_info.to_id();
+
+        ledger.delegation.register_stake_pool(pool_info).unwrap();
+
+        (pool_id, pool_vrf_private_key)
+    }
+
+    #[test]
+    #[ignore]
+    pub fn test_phi() {
+        let slots_per_epoch = 200000;
+        let active_slots_coeff = 0.1;
+
+        let (_genesis_hash, mut ledger) = create_initial_fake_ledger(
```

Comment:
if this is not used, I think you can also just do `_`

Reply:
It's not used. marking it as _genesis_hash is done to both signify it is not used (so it doesn't trip the rustc unused variable linter), while also still documenting what it is we're iterating over.

This PR is also just updating our fork with IOHK's repo (thus this is their code) before we send them a PR back, so we don't want to be making needless changes which would just complicate our PRs in their repo.

All changes in this PR are also in the in-development/experimental shelley module which we are not using yet for the bridge or anything else.